### PR TITLE
Mobile Worker Self-Registration

### PR DESCRIPTION
**Pull request by gcapalbo**
Creates a workflow for letting mobile workers self-register.

https://docs.google.com/document/d/1s2q3Nmiwh_SJm9S1O16twbt3C1JCzEUy6EGpLITCOEc/edit#heading=h.q3wyha58n4s0

@NoahCarnahan code review buddy
@krsdimagi fyi

### CODE REVIEW

File: corehq/apps/sms/urls.py [hunk 1]
```diff
@@ -38,6 +39,7 @@
     url(r'^languages/edit/$', 'edit_sms_languages', name='edit_sms_languages'),
     url(r'^translations/download/$', 'download_sms_translations', name='download_sms_translations'),
     url(r'^translations/upload/$', 'upload_sms_translations', name='upload_sms_translations'),
+    url(r'^invitations/$', ManageRegistrationInvitationsView.as_view(), name=ManageRegistrationInvitationsView.urlname),
```
Comment (dimagimon) [inline corehq/apps/sms/urls.py]: E501 line too long (120 > 115 characters)

File: requirements/requirements.txt [hunk 2]
```diff
@@ -33,6 +33,7 @@ phonenumberslite==6.0.0a
 poster==0.8.1
 psycopg2==2.5.1
 psycogreen==1.0
+pycrypto==2.6.1
```
Comment (gcapalbo) [inline requirements/requirements.txt]: Looks like we already have pycrypto installed in our codebase (must be a dependency for another package), but just pinning it here

**Comment by krsdimagi** [comment]
@gcapalbo wow, this looks absolutely incredible!

**Comment by gcapalbo** [comment]
@biyeun new code review buddy

I wasn't aware that new features should be using bootstrap 3, so this is in bootstrap 2. We're kind of crunched for time on this one, so I don't think I can update this PR, but I'm happy to do the update in a separate PR.

Also, it doesn't look like we have a bootstrap 3 template for mobile devices like we do for bootstrap 2 (`style/bootstrap2/base_mobile.html`).  (One of the pages this feature creates is a page that's intended for use only on mobile devices)

**Comment by biyeun** [comment]
@gcapalbo would definitely be a good idea to move to bootstrap3 in a separate PR, and if you need help with this we should try remote pairing.

Bootstrap 3 is super responsive and designed for mobile, so this would be the perfect usecase for that. Bootstrap 2 really sucks at responsive design, and is one of the reasons why support moved entirely over to Bootstrap 3.

File: corehq/apps/app_manager/dbaccessors.py [hunk 3]
```diff
@@ -105,3 +105,19 @@ def get_apps_in_domain(domain, full=False, include_remote=True):
...
+    Returns the app ids of all apps in the domain that have at least one build.
+    """
+    from .models import Application
+    result = Application.get_db().view(
+        'app_manager/saved_app',
+        startkey=[domain],
+        endkey=[domain, {}],
+        include_docs=False,
+    )
+    app_ids = [data.get('value', {}).get('copy_of') for data in result]
```
Comment (biyeun) [inline corehq/apps/app_manager/dbaccessors.py]: could this include blank app_ids?

File: corehq/apps/sms/views.py [hunk 4]
```diff
@@ -1616,3 +1628,140 @@ def post(self, request, *args, **kwargs):
...
+        return context
+
+    @property
+    def total(self):
+        return SelfRegistrationInvitation.objects.filter(domain=self.domain).count()
+
+    @property
+    def column_names(self):
+        return [
+            'Created On',
```
Comment (biyeun) [inline corehq/apps/sms/views.py]: translations?

File: corehq/apps/users/templates/users/mobile/commcare_user_self_register.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{% extends "style/bootstrap2/base_mobile.html" %}
+{% load hq_shared_tags %}
+{% load crispy_forms_tags %}
+
+{% block title %}CommCareHQ Registration{% endblock %}
+
+{% block content %}
+<div class="page-header">
+    <h1>CommCareHQ Registration</h1>
```
Comment (biyeun) [inline corehq/apps/users/templates/users/mobile/commcare_user_self_register.html]: translations?

**Comment by biyeun** [comment]
@gcapalbo left a few comments, otherwise looks good. Also made a shiny FB ticket for you! http://manage.dimagi.com/default.asp?182630 ;)

File: corehq/apps/app_manager/dbaccessors.py (hunk 3 above)
Comment (gcapalbo) [inline corehq/apps/app_manager/dbaccessors.py]: see two lines below - any blank app_ids are filtered out

**Comment by gcapalbo** [comment]
@biyeun updated, should be g2g

Also, good to know about bootstrap 3 support for mobile. Do we need a separate base template like we have for bootstrap 2 (`style/bootstrap2/base_mobile.html`)?

**Comment by biyeun** [comment]
@gcapalbo yes, we should have a separate base template for that---how soon will you be able to address that ticket?

**Comment by gcapalbo** [comment]
@biyeun I probably won't be able to get to it until the next roadmap cycle

**Pull request merged**